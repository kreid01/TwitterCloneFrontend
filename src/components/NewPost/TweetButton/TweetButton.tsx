import React from 'react'

interface Props {
    handleTweet: () => void
    label:string
}

export const TweetButton: React.FC<Props> = ({ handleTweet, label }) => {
    
    return (
        <button
        data-testid="tweetButton"
        onClick={handleTweet}
        className='font-bold text-white ml-auto 
        bg-blue-500 hover:bg-blue-400 p-2 pl-4
         pr-4 rounded-2xl'
        >{label}</button>
    )
}