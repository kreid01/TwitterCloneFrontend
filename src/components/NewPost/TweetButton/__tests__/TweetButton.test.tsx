/* eslint-disable testing-library/prefer-screen-queries */
import React from 'react'
import { TweetButton} from '../TweetButton'

import { screen, render, cleanup, fireEvent } from '@testing-library/react';
import "@testing-library/jest-dom";

import renderer from 'react-test-renderer'

const handleTweet = jest.fn()

afterEach(cleanup)
it("render without crashing", () => {
    render(<TweetButton
    label='Tweet'
    handleTweet={handleTweet}/>)
})

it("renders button correctly", () => {
    render(<TweetButton
    label='Tweet'
    handleTweet={handleTweet} />)
    const element =  screen.getByTestId("tweetButton")
    expect(element).toHaveTextContent('Tweet')
})

it("calls handleTweeet function when clicked", () => {
    render(<TweetButton
        label='Tweet'
        handleTweet={handleTweet} />) 
    fireEvent.click(screen.getByTestId("tweetButton"))
    expect(handleTweet).toHaveBeenCalled()
})
 
it("matched snapshot", () => {
    const tree = renderer.create(<TweetButton handleTweet={handleTweet}
    label="Tweet"/>).toJSON()
    expect(tree).toMatchSnapshot()
})
