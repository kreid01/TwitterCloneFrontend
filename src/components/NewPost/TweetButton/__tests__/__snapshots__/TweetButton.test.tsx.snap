// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matched snapshot 1`] = `
<button
  className="font-bold text-white ml-auto 
 bg-blue-500 hover:bg-blue-400 p-2 pl-4
 pr-4 rounded-2xl"
  data-testid="tweetButton"
  onClick={[MockFunction]}
>
  Tweet
</button>
`;
